DeepSpeech.py

```diff
@@ -491,156 +428,140 @@ def end(self, session):
         dropout_rates[5]: 0.,
     }
 
-    # Progress Bar
-    def update_progressbar(set_name):
-        if not hasattr(update_progressbar, 'current_set_name'):
-            update_progressbar.current_set_name = None
-
-        if (update_progressbar.current_set_name != set_name or
-            update_progressbar.current_job_index == update_progressbar.total_jobs):
-
-            # finish prev pbar if it exists
-            if hasattr(update_progressbar, 'pbar') and update_progressbar.pbar:
-                update_progressbar.pbar.finish()
+    # Building the graph
+    optimizer = create_optimizer()
+    gradients, loss = get_tower_results(model_feeder, optimizer, dropout_rates)
+    # Average tower gradients across GPUs
+    avg_tower_gradients = average_gradients(gradients)
+    log_grads_and_vars(avg_tower_gradients)
+    # global_step is automagically incremented by the optimizer
+    global_step = tf.Variable(0, trainable=False, name='global_step')
+    apply_gradient_op = optimizer.apply_gradients(avg_tower_gradients, global_step=global_step)
 
-            update_progressbar.total_jobs = None
-            update_progressbar.current_job_index = 0
+    # Summaries
+    step_summaries_op = tf.summary.merge_all('step_summaries')
+    step_summary_writers = {
+        'train': tf.summary.FileWriter(os.path.join(FLAGS.summary_dir, 'train'), max_queue=120),
+        'dev': tf.summary.FileWriter(os.path.join(FLAGS.summary_dir, 'dev'), max_queue=120)
+    }
 
-            current_epoch = coord._epoch-1
+    # Checkpointing
+    epoch_saver = tf.train.Saver(max_to_keep=FLAGS.max_to_keep)
```

Comment:
Previously we had checkpoints saved every 10 minutes so that we could recover from a crash without losing too much training time. This is now only saving on every epoch end. Saving checkpoints at the end of epochs is nice to have, but I think saving every 10 minutes is a good safety feature to keep.

Reply:
Will address this...